### PR TITLE
Storage db

### PR DESCRIPTION
**Pull request by hubert-groen**
- storage endpoints
- storage database in sqlite
- notification endpoints
- notification database in sqlite
- payment (for now only single endpoint)

### CODE REVIEW

File: backend/storage/storage_main.py [hunk 1]
```diff
@@ -1,8 +1,83 @@
...
+
+@storage_app.get("/user")
+async def get_user(token: Annotated[str, Depends(verify_token)]) -> UserModel:
+    storage_repo = StorageRepository()
+    user_id=token["sub"]
+    # user_id += "1"
+    try:
+        user = storage_repo.get_user(user_id=user_id)
+    except:
+        raise HTTPException(status_code=400, detail=f"User does not exists!")
```
Comment (pkosmala01) [inline backend/storage/storage_main.py:38]:
```suggestion
        raise HTTPException(status_code=400, detail=f"User does not exist!")
```

File: backend/storage/storage_main.py [hunk 2]
```diff
@@ -1,8 +1,83 @@
...
+
+@storage_app.get("/files")
+async def get_files_for_user(token: Annotated[str, Depends(verify_token)]) -> list[FileModel]:
+    storage_repo = StorageRepository()
+    user_id=token["sub"]
+    # user_id += "1"
+    try:
+        files = storage_repo.get_files(user_id=user_id)
+    except:
+        raise HTTPException(status_code=400, detail=f"User does not exists!")
```
Comment (pkosmala01) [inline backend/storage/storage_main.py:62]:
```suggestion
        raise HTTPException(status_code=400, detail=f"User does not exist!")
```

File: backend/storage/storage_main.py [hunk 3]
```diff
@@ -1,8 +1,83 @@
...
+    return files
+
+
+@storage_app.patch("/files")
+async def rename_file(file_id: uuid.UUID, file_rename: FileRenameModel, token: Annotated[str, Depends(verify_token)]) -> FileIdResponse:
+    storage_repo = StorageRepository()
+    try:
+        updated_file_id = storage_repo.rename_file(file_id=file_id, new_file=file_rename)
+    except:
+        raise HTTPException(status_code=400, detail=f"File does not exists!")
```
Comment (pkosmala01) [inline backend/storage/storage_main.py:72]:
```suggestion
        raise HTTPException(status_code=400, detail=f"File does not exist!")
```

File: backend/storage/storage_main.py [hunk 4]
```diff
@@ -1,8 +1,83 @@
...
+
+
+@storage_app.get("/user")
+async def get_user(token: Annotated[str, Depends(verify_token)]) -> UserModel:
+    storage_repo = StorageRepository()
+    user_id=token["sub"]
+    # user_id += "1"
+    try:
+        user = storage_repo.get_user(user_id=user_id)
+    except:
```
Comment (pkosmala01) [inline backend/storage/storage_main.py:37]: I'm not going to leave this comment everywhere, but please, always specify exactly what exception/exceptions you want to catch

File: backend/pyproject.toml [hunk 5]
```diff
@@ -2,7 +2,8 @@
 name = "backend"
 description = ""
 readme = "README.md"
-package-mode = false
+authors = []
+version = "0.0.1"
```
[the remark this answers is not in the document]
Comment (hubert-groen) [reply backend/pyproject.toml:6]: Do you mean to simply delete these 2 lines?

File: backend/storage/repository.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+from storage.database_definition import UserTable, FileTable
+from storage.exceptions import UserAlreadyExists, UserDoesNotExists, FileAlreadyExists, FileDoesNotExists
+
+import logging
+
+
+class StorageRepository():
+    def __init__(self) -> None:
+        engine = create_engine("sqlite:///storage/storage.db")
+        self._connection = engine.connect()
```
[the remark this answers is not in the document]
Comment (hubert-groen) [reply backend/storage/repository.py]: done

File: backend/storage/repository.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            raise UserDoesNotExists()
+        finally:
+            self._connection.close()
+        return UserModel.model_validate(user)
+
+    def insert_file(self, file: FileModel) -> uuid.UUID:
+        # FIXME: miałem tego nie robić w ten sposób,
+        # ale file_name nie jest primary key, więc akceptuje duplikaty nazw i nie oznacza to błędu
+        stmt_check = select(FileTable).where(FileTable.c.file_name == file.file_name)
+        result = self._connection.execute(stmt_check).fetchone()
```
[the remark this answers is not in the document]
Comment (hubert-groen) [reply backend/storage/repository.py]: ok, I'm erasing this

File: backend/main.py [hunk 8]
```diff
@@ -49,6 +53,11 @@ def exception_handler_faculty(request: Request, exc: RedirectNotificationService
     return RedirectResponse(url='http://localhost:8000/notification/hello')
```
[the remark this answers is not in the document]
Comment (hubert-groen) [reply backend/main.py:58]: done

File: backend/storage/storage_main.py (hunk 4 above)
Comment (hubert-groen) [reply backend/storage/storage_main.py]: Sure. I placed except IntegrityError everywhere, is that ok?

Comment (pkosmala01) [reply backend/storage/storage_main.py]: Almost, in endpoints we should be intercepting our custom exceptions that are raised in the repositories

**Pull request merged**